contrib/src/main/java/com/datatorrent/contrib/parquet/ParquetFilePOJOReader.java

```diff
@@ -0,0 +1,318 @@
+ * of now only primitive types ( INT32, INT64, BOOLEAN, FLOAT, DOUBLE, BINARY )
+ * are supported.
+ *
+ * Parquet primitive types maps to POJO datatypes as follows : INT32 -> int
+ * INT64 -> long BOOLEAN -> boolean FLOAT -> float DOUBLE -> double BINARY ->
+ * String
+ *
+ * @displayName ParquetFilePOJOReader
+ * @tags parquet,input adapter
+ * @since 3.3.0
+ */
+public class ParquetFilePOJOReader extends AbstractParquetFileReader<Object>
+{
+
+  /**
+   * POJO class
+   */
+  protected transient Class<?> pojoClass;
+  /**
+   * Map containing setters for fields in POJO
+   */
+  protected transient Map<String, Setter> pojoSetters;
+  /**
+   * String representing Parquet TO POJO field mapping. If not provided, then
+   * reflection is used to determine the mapping. Format :
+   * PARQUET_FIELD_NAME:POJO_FIELD_NAME:TYPE
+   * E.g.event_id:event_id_v2:INTEGER,org_id:org_id_v2:STRING,long_id:
+   * long_id_v2:
+   * LONG,css_file_loaded:css_file_loaded_v2:BOOLEAN,float_val:float_val_v2:
+   * FLOAT,double_val:double_val_v2:DOUBLE
+   */
+  protected transient String groupToPOJOFieldsMapping = null;
+  protected transient List<FieldInfo> fieldInfos;
+  protected transient List<ActiveFieldInfo> columnFieldSetters = null;
+  protected static final String FIELD_SEPARATOR = ":";
+  protected static final String RECORD_SEPARATOR = ",";
+  private static final Logger logger = LoggerFactory.getLogger(ParquetFilePOJOReader.class);
+
+  @OutputPortFieldAnnotation(schemaRequired = true)
+  public final transient DefaultOutputPort<Object> output = new DefaultOutputPort<Object>()
+  {
+
+    @Override
+    public void setup(Context.PortContext context)
+    {
+      pojoClass = context.getValue(Context.PortContext.TUPLE_CLASS);
+      pojoSetters = Maps.newHashMap();
+      for (Field f : pojoClass.getDeclaredFields()) {
+        try {
+          pojoSetters.put(f.getName(), generateSettersForField(pojoClass, f.getName()));
+        } catch (NoSuchFieldException | SecurityException e) {
+          logger.error("Failed to generate setters. Exception {}", e);
+          throw new RuntimeException(e);
+        }
+      }
+
+      if (groupToPOJOFieldsMapping == null) {
+        fieldInfos = createFieldInfoMap(generateFieldInfoInputs());
+      } else {
+        fieldInfos = createFieldInfoMap(groupToPOJOFieldsMapping);
+      }
+      initColumnFieldSetters();
+    }
+
+  };
+
+  /**
+   * Converts Group to POJO
+   */
+  @Override
+  protected Object convertGroup(Group group)
+  {
+    Object obj;
+    try {
+      obj = pojoClass.newInstance();
+    } catch (InstantiationException | IllegalAccessException ex) {
+      throw new RuntimeException(ex);
+    }
+    for (int i = 0; i < columnFieldSetters.size(); i++) {
+      try {
+        ParquetFilePOJOReader.ActiveFieldInfo afi = columnFieldSetters.get(i);
+        int fieldIndex = schema.getFieldIndex(afi.fieldInfo.getColumnName());
```

Comment:
With above aproach, you even don't need to get fieldIndex using schema.getFieldIndex everytime.
Your array structure has it.